crates/proof-of-sql/src/sql/ast/group_by_expr.rs

```diff
@@ -141,11 +144,28 @@ impl<C: Commitment> ProofExpr<C> for GroupByExpr<C> {
                 sum_result_columns_evals,
                 count_column_eval,
             ),
-        )
-
-        // todo!: check that the group_by results are unique.
-        //        When the GroupByExpr is the root node of the Proof plan,
-        //        this can be done by simply looking at the results returned by the prover.
+        )?;
+        match result {
+            Some(table) => {
+                let cols = self
+                    .group_by_exprs
+                    .iter()
+                    .map(|col| table.inner_table().get(&col.column_id()))
+                    .collect::<Option<Vec<_>>>()
+                    .ok_or(ProofError::VerificationError(
+                        "Result does not all correct group by columns.",
+                    ))?;
```

Comment:
Not that we need to fix this now but we really need to split out `VerificationError` into numerous variants based on what the actual error is.